cgogn/core/cmap/map_base.h

```diff
@@ -617,79 +618,84 @@ class MapBase : public MapBaseData<MAP_TRAITS>
 
 protected:
 
+	/*!
+	 * \Brief Methods to iterate over darts with a MASK that filters the traversed darts
+	 * A MASK is an functor that determine if a dart should be traversed or skipped.
+	 * It return false when a dart should be skipped, true in other cases.
+	 */
 	template <typename MASK>
-	class const_iterator
+	inline Dart begin(const MASK& mask) const
 	{
-	public:
-
-		const Self& map_;
-		const MASK& mask_;
-		Dart dart_;
-
-		inline const_iterator(const Self& map, Dart d, const MASK& mask) :
-			map_(map),
-			dart_(d),
-			mask_(mask)
-		{}
-
-		inline const_iterator(const const_iterator& it) :
-			map_(it.map_),
-			dart_(it.dart_),
-			mask_(it.mask_)
-		{}
-
-		inline const_iterator& operator=(const const_iterator& it)
-		{
-			map_ = it.map_;
-			dart_ = it.dart_;
-			mask_ = it.mask_;
-			return *this;
+		if (check_func_return_void(MASK)) {
```

Comment:
J'ai trouvé, je commit et pus après manger 😃

Sur la gestion du end() les avis divergent.
Je propose qu'on en parle de vive voix.

Mais si tu veux que la boucle se poursuivent sur des brins ajoutés, il
suffit d'évaluer end() à chaque fois, sans le stocker à l'initialisaton.
Le 18 mars 2016 11:41, "Sylvain Thery" <notifications@github.com> a écrit :

> In cgogn/core/cmap/map_base.h
> <https://github.com/cgogn/CGoGN_2/pull/149#discussion_r56638668>:
>
> > -			mask_(mask)
> > -		{}
> > -
> > -		inline const_iterator(const const_iterator& it) :
> > -			map_(it.map_),
> > -			dart_(it.dart_),
> > -			mask_(it.mask_)
> > -		{}
> > -
> > -		inline const_iterator& operator=(const const_iterator& it)
> > -		{
> > -			map_ = it.map_;
> > -			dart_ = it.dart_;
> > -			mask_ = it.mask_;
> > -			return *this;
> > +		if (check_func_return_void(MASK)) {
>
> Et pourquoi pas plutôt écrire les version de begin et next sans le
> paramètre de mask:
>
> inline Dart begin() const
> {
>     Dart d = Dart(this->topology_.begin());
> }
>
> inline void next(Dart& d) const
> {
>     Dart end = Dart(this->topology_.end());
>     do {
>         this->topology_.next(d.index);
>     } while (d != end);
> }
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cgogn/CGoGN_2/pull/149/files/313baacaeb7883dc4d751e02d577605f6d6ec62c#r56638668>
>